### PR TITLE
MSSQL '?' solution

### PR DESCRIPTION
**Pull request by KalebMatthews**
Now escapes properly if literal '?' is needed.

Change credit to @caihaibin1991 (unable to link).

```
Coded and tested. '?' will be replaced with @p# while '\?' will remain as '?'.
```

### CODE REVIEW

**Comment by elhigu** [comment]
Missing regression tests

**Comment by caihaibin1991** [comment]
@KalebMatthews thank you

File: test/unit/dialects/mssql.js [hunk 1]
```diff
@@ -30,4 +30,26 @@ describe('MSSQL unit tests', () => {
       'select * from [projects] where [id" = 1 UNION SELECT 1, @@version -- --] = ?'
     );
   });
+
+  it('should include "?" in statement', async () => {
```
Comment (elhigu) [inline test/unit/dialects/mssql.js:34]: test naming is pretty cryptic, maybe following would be more clear about tests intent:

`should not convert escaped "?" -marks in raw SQL to native parameter bindings` 

the same applies to second test:

`should convert "?" -marks in raw SQL to native parameter binding`

**Comment by kibertoad** [comment]
@KalebMatthews Could you please pull latest changes from master?

**Comment by kibertoad** [comment]
Thank you!

**Pull request merged**